backend/src/agent/persistence/client/pipeline_client.go

```diff
@@ -96,6 +105,15 @@ func (p *PipelineClient) ReportWorkflow(workflow util.ExecutionSpec) error {
 				statusCode.Message(),
 				err.Error(),
 				workflow.ToStringForStore())
+		} else if statusCode.Code() == codes.Unauthenticated && strings.Contains(err.Error(), "service account token has expired") {
+			//if unauthenticated because SA token is expired, re-read/refresh the token and try again
+			p.tokenRefresher.RefreshToken()
+			return util.NewCustomError(err, util.CUSTOM_CODE_TRANSIENT,
```

Comment:
Although I see you're following an existing pattern, for my education how would user discover such error? Would this be surfaced back by apiserver?

Reply:
Hi! Sorry for my late response. 
The user could observe the status of the Run and inspect the logs of the persistent agent. The errors are not directly displayed on the UI. 
If a "transient" error occurs the Persistent Agent will retry with an exponential delay

Reply:
Hi @chensun 
I have added unit tests for the TokenRefresher. For the need of the unit tests I created the `FileReader interface`.

Reply:
Here, If an error occurs while refreshing/reading the SA token I just log the error and let the worker thread retry. Any idea how to handle this gracefully?